h2o-r/h2o-package/R/export.R

```diff
@@ -240,12 +241,13 @@ h2o.save_frame <- function(x, dir, force = TRUE) {
 #' # h2o.saveModel(object = prostate_glm, path = "/Users/UserName/Desktop", force = TRUE)
 #' }
 #' @export
-h2o.saveModel <- function(object, path="", force=FALSE, export_cross_validation_predictions=FALSE) {
+h2o.saveModel <- function(object, path="", force=FALSE, export_cross_validation_predictions=FALSE, filename=object@model_id) {
```

Comment:
wouldn't this fail in a non-obvious way if object is not actual H2OModel?

it basically means the validation doesn't show the error message if the object is not what is expected

Reply:
True, thanks, I have updated all save and download methods and add test to catch this case 👍